cloudfunction/publish_metrics/main.py

```diff
@@ -0,0 +1,19 @@
+import base64
+import functions_framework
+import os
+
+from cloudevents.http import CloudEvent
+from google.cloud import pubsub_v1
+
+@functions_framework.cloud_event
+def publish_metrics(cloud_event: CloudEvent) -> None:
+    destination_project_id = os.getenv('TARGET_GOOGLE_CLOUD_PROJECT')
+    destination_topic_name = os.getenv('TARGET_TOPIC_NAME')
+
+    pubsub_message = base64.b64decode(cloud_event.data["message"]["data"]).decode()
+    attributes = cloud_event.data["message"]["attributes"]
+
+    publisher = pubsub_v1.PublisherClient()
+    destination_topic_path = f'projects/{destination_project_id}/topics/{destination_topic_name}'
```

Comment:
How does the authentication happens? can you write some readme in this folder with a guideline how to deploy and authenticate to publish the message to the target topic?

Reply:
No need for authentication, As this will be deployed in the data plane, we will only allow ingress from the same VPC. furthermore the trigger for the Cloudfunction will be set to the Data-Plane Pub/Sub topic. It won't be triggered by anything else.